src/response.rs

```diff
@@ -61,17 +62,48 @@ impl Response {
         }
     }
 
-    pub fn records(&self) -> ::std::vec::IntoIter<&Record> {
-        let records: Vec<_> = vec![&self.answers, &self.nameservers, &self.additional]
-            .into_iter()
-            .flat_map(|c| c.iter())
-            .collect();
-        records.into_iter()
+    pub fn records(&self) -> impl Iterator<Item = &Record> {
+        self.answers
+            .iter()
+            .chain(self.nameservers.iter())
+            .chain(self.additional.iter())
     }
 
     pub fn is_empty(&self) -> bool {
         self.answers.is_empty() && self.nameservers.is_empty() && self.additional.is_empty()
     }
+
+    pub fn get_ip_addr(&self) -> Option<IpAddr> {
+        self.records().find_map(|record| match record.kind {
+            RecordKind::A(addr) => Some(addr.into()),
+            RecordKind::AAAA(addr) => Some(addr.into()),
+            _ => None,
+        })
+    }
+
+    pub fn get_hostname(&self) -> Option<&str> {
+        self.records().find_map(|record| match record.kind {
+            RecordKind::PTR(ref host) => Some(host.as_str()),
+            _ => None,
+        })
+    }
+
+    pub fn get_port(&self) -> Option<u16> {
+        self.records().find_map(|record| match record.kind {
+            RecordKind::SRV { port, .. } => Some(port),
+            _ => None,
+        })
+    }
```

Comment:
Recommend adding a `socket_address` method that returns a `std::net::SocketAddr` - this contains both the hostname and port, and can also be convienently given to the socket connection methods because it necessarily implements `ToSocketAddrs`.

This should also make the debug printing in the chromecast example slightly cleaner

rather than

```rust
        let addr = response.get_ip_addr();
        let host = response.get_hostname();

        if let (Some(host), Some(addr)) = (host, addr) {
            println!("found cast device {} at {}", host, addr);
```

it'd be

```rust
        if let Some(addr) = response.socket_address() {
            println!("found cast device at {}", addr);
```